server/src/main/java/org/apache/druid/server/metrics/GroupByStatsMonitor.java

```diff
@@ -41,21 +43,28 @@ public class GroupByStatsMonitor extends AbstractMonitor
 {
   private final GroupByStatsProvider groupByStatsProvider;
   private final BlockingPool<ByteBuffer> mergeBufferPool;
+  private final Map<String, String[]> dimensions;
```

Comment:
Maybe rename to `serviceDimensions`?

In fact, I wonder if it wouldn't be cleaner to just bind the datasource and taskID as `@ExtraServiceDimensions` so that they are used in `EmitterModule` and are emitted for all metrics emitted by a peon.

Reply:
> I can rename, but I left this field as dimensions to be consistent with the other monitors.

I see, thanks for the clarification. Could you point me to some of these monitors?